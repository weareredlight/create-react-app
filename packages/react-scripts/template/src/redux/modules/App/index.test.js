import configureMockStore from 'redux-mock-store';

import AppReducer, { addName, removeName, getPost } from './index';

const mockStore = configureMockStore([]);
const store = mockStore({});

beforeEach(() => {
  store.clearActions();
});

// Override system functions to garantee always the same result
global.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        userId: 1,
        id: 1,
        title: 'sunt aut facere repellat provident occaecati',
        body:
          'quia et suscipit suscipit recusandae consequuntur rem eveniet architecto',
      }),
  }),
);

const NEW_DATE = new Date('2017');
global.Date = jest.fn(() => NEW_DATE);

// Test Actions
it('[Action] ADD_NAME (Rui Miguel)', () => {
  store.dispatch(addName('Rui Miguel'));
  expect(store.getActions()).toMatchSnapshot();
});

it('[Action] ADD_NAME (empty)', () => {
  store.dispatch(addName(''));
  expect(store.getActions()).toMatchSnapshot();
});

it('[Action] SUBTRACT_NAME (2)', () => {
  store.dispatch(removeName(2));
  expect(store.getActions()).toMatchSnapshot();
});

it('[Action] SUBTRACT_NAME (empty)', () => {
  store.dispatch(removeName());
  expect(store.getActions()).toMatchSnapshot();
});

it('[Action] GET_POST', async () => {
  store.dispatch(await getPost());
  expect(store.getActions()).toMatchSnapshot();
});

// Test Reducers
it('[Reducer] INITIAL STATE', () => {
  const action = { type: 'DUMMY' };
  expect(AppReducer(undefined, action)).toMatchSnapshot();
});

it('[Reducer] ADD_NAME (Rui Miguel)', () => {
  expect(AppReducer(undefined, addName('Rui Miguel'))).toMatchSnapshot();
});

it('[Reducer] SUBTRACT_NAME (3)', () => {
  expect(AppReducer(undefined, removeName(3))).toMatchSnapshot();
});

it('[Reducer] GET_POST', async () => {
  expect(AppReducer(undefined, await getPost())).toMatchSnapshot();
});
