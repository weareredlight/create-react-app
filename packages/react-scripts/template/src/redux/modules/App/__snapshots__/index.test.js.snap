// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Action] ADD_NAME (Rui Miguel) 1`] = `
Array [
  Object {
    "payload": Object {
      "name": "Rui Miguel",
    },
    "type": "ADD_NAME",
  },
]
`;

exports[`[Action] ADD_NAME (empty) 1`] = `
Array [
  Object {
    "payload": Object {
      "name": "",
    },
    "type": "ADD_NAME",
  },
]
`;

exports[`[Action] GET_POST 1`] = `
Array [
  Object {
    "payload": Object {
      "post": Object {
        "body": "quia et suscipit suscipit recusandae consequuntur rem eveniet architecto",
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati",
        "userId": 1,
      },
    },
    "type": "GET_POST",
  },
]
`;

exports[`[Action] SUBTRACT_NAME (2) 1`] = `
Array [
  Object {
    "payload": Object {
      "index": 2,
    },
    "type": "SUBTRACT_NAME",
  },
]
`;

exports[`[Action] SUBTRACT_NAME (empty) 1`] = `
Array [
  Object {
    "payload": Object {
      "index": undefined,
    },
    "type": "SUBTRACT_NAME",
  },
]
`;

exports[`[Reducer] ADD_NAME (Rui Miguel) 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
    Object {
      "id": 1483228800000,
      "name": "Rui Miguel",
    },
  ],
  "post": null,
}
`;

exports[`[Reducer] GET_POST 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
  ],
  "post": Object {
    "body": "quia et suscipit suscipit recusandae consequuntur rem eveniet architecto",
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati",
    "userId": 1,
  },
}
`;

exports[`[Reducer] INITIAL STATE 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
  ],
  "post": null,
}
`;

exports[`[Reducer] SUBTRACT_NAME (3) 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
  ],
  "post": null,
}
`;
